# Introducción al análisis: filtrar y ordenar datos



### 1. Importa las librerías necesarias

In [101]:
import numpy as np
import pandas as pd

### 2. Importa el dataset de esta dirección [address](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv) y guardarlo en una variable llamada chipo.

In [102]:
df= pd.read_csv("https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv", delimiter= '\t')
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

### 3. Cuál es el producto más valioso? Piénsalo bien.

In [103]:
df['item_price']= df['item_price'].str.replace("$", "")


In [104]:
df['item_price']

0        2.39 
1        3.39 
2        3.39 
3        2.39 
4       16.98 
         ...  
4617    11.75 
4618    11.75 
4619    11.25 
4620     8.75 
4621     8.75 
Name: item_price, Length: 4622, dtype: object

In [105]:
df['item_price']= df['item_price'].astype(float)

In [106]:
df['quantity']= df['quantity'].astype(float)

In [107]:
df['valor_prod']= df['item_price']/ df['quantity']

In [108]:
valor_prod= df['item_price']/ df['quantity']

In [109]:
df['valor_prod'].argmax()

281

In [110]:
valor_prod[valor_prod.argmax()]

11.89

In [111]:
df.iloc[281]

order_id                                                            123
quantity                                                            2.0
item_name                                              Steak Salad Bowl
choice_description    [Tomatillo Red Chili Salsa, [Black Beans, Chee...
item_price                                                        23.78
valor_prod                                                        11.89
Name: 281, dtype: object

ESTE ESTA BIEN HECHO

Otra forma: 

df[df['valor_prod'] == df['valor_prod'].max()]['item_name].unique()


In [112]:
df[df['valor_prod'] == df['valor_prod'].max()]['item_name'].unique()


array(['Steak Salad Bowl', 'Carnitas Salad Bowl', 'Barbacoa Salad Bowl'],
      dtype=object)

### 4. Cuántos productos cuestan más de $10.00?

In [113]:
prods_mayor_diez = np.array(valor_prod >= 10.0)

In [114]:
df['valor_prod']= df['valor_prod'].astype(float)

In [115]:
np.sum(df['valor_prod'] > 10)

1034

In [116]:
len(df['valor_prod'][df['valor_prod']> 10])

1034

In [117]:
nuevo_df= df[df.valor_prod > 10].copy()

In [118]:
# 1034 productos

CORRECCION: ESTÁ MAL

nombres= df[df['valor_prod']> 10]['item_name'].unique()
no_pedidos= df[df['valor_prod']> 10]['item_name'].nunique()

In [119]:
df[df['valor_prod']> 10]['item_name'].nunique()

25

CORRRECCION: 25 productos

### 4.1: Y cuántos pedidos se han hecho con un producto de más de 10$?

In [120]:
nuevo_df

order_id  quantity           item_name  \
5            3       1.0        Chicken Bowl   
7            4       1.0       Steak Burrito   
13           7       1.0        Chicken Bowl   
23          12       1.0     Chicken Burrito   
39          19       1.0       Barbacoa Bowl   
...        ...       ...                 ...   
4610      1830       1.0       Steak Burrito   
4611      1830       1.0      Veggie Burrito   
4617      1833       1.0       Steak Burrito   
4618      1833       1.0       Steak Burrito   
4619      1834       1.0  Chicken Salad Bowl   

                                     choice_description  item_price  \
5     [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98   
7     [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.75   
13    [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25   
23    [[Tomatillo-Green Chili Salsa (Medium), Tomati...       10.98   
39    [Roasted Chili Corn Salsa, [Fajita Vegetables,...       11.75   
...                                                 ...         ...   
4610  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75   
4611  [Tomatillo Green Chili Salsa, [Rice, Fajita Ve...       11.25   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25   

      valor_prod  
5          10.98  
7          11.75  
13         11.25  
23         10.98  
39         11.75  
...          ...  
4610       11.75  
4611       11.25  
4617       11.75  
4618       11.75  
4619       11.25  

[1034 rows x 6 columns]

In [121]:
nuevo_df['order_id'].nunique()

787

In [122]:
# 787 pedidos tienen un producto de mas de 10€

ESTE ESTÁ BIEN

### 4.2: Y cuántos pedidos se han hecho de más de 10$?

In [123]:
nuevo_df2= df[df.item_price > 10].copy()


In [124]:
nuevo_df2

order_id  quantity           item_name  \
4            2       2.0        Chicken Bowl   
5            3       1.0        Chicken Bowl   
7            4       1.0       Steak Burrito   
13           7       1.0        Chicken Bowl   
23          12       1.0     Chicken Burrito   
...        ...       ...                 ...   
4610      1830       1.0       Steak Burrito   
4611      1830       1.0      Veggie Burrito   
4617      1833       1.0       Steak Burrito   
4618      1833       1.0       Steak Burrito   
4619      1834       1.0  Chicken Salad Bowl   

                                     choice_description  item_price  \
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98   
5     [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98   
7     [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.75   
13    [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25   
23    [[Tomatillo-Green Chili Salsa (Medium), Tomati...       10.98   
...                                                 ...         ...   
4610  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75   
4611  [Tomatillo Green Chili Salsa, [Rice, Fajita Ve...       11.25   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25   

      valor_prod  
4           8.49  
5          10.98  
7          11.75  
13         11.25  
23         10.98  
...          ...  
4610       11.75  
4611       11.25  
4617       11.75  
4618       11.75  
4619       11.25  

[1130 rows x 6 columns]

In [125]:
nuevo_df2['order_id'].nunique()

863

In [126]:
#se han hecho 863 pedidos de mas de 10€

CORRECCION: ESTA MAL

primero: cuanto se ha pagado por pedido? vamos a agrupar por pedido
segundo: lista de pedidos que han sido de mas de diez dolares
tercero: cual es la longitud de esa lista?

In [127]:
orders_sum= df.groupby("order_id")['item_price'].sum()

In [128]:
orders_sum_mayor_10= orders_sum[orders_sum>10]

In [129]:
len(orders_sum_mayor_10)

1834

CORRECCION: se han hecho 1834 pedidos de mas de 10 $ 

### 4.3: Y en cuántos pedidos se ha pagado más de 10$ por un mismo producto?

In [130]:
nuevo_df['order_id'].nunique()

787

CORRECCION: ESTA MAL, Da la impresion de que lo he hecho en el 4.2. Es posble que haya planteado mal la respuesta.

In [131]:
df[df['item_price']>10]['order_id'].nunique()

863

CORRECCION: En 863 pedidos se ha pagado mas de 10 $ por un mismo producto (por ej. dos botellas de vino de 6 $ en un mismo pedido)

### 5. Ordena de mayor a menor por el nombre del producto y por precio unitario

In [132]:
nombre_items= df['item_name'].unique()

In [133]:

lista_valores=[]
for i in range(len(nombre_items)):
    lista_valores.append(df['valor_prod'][i])


lista_nombres=[]
for i in range(len(nombre_items)):
    lista_nombres.append(df['item_name'][i])


In [134]:
print(lista_valores)

print(nombre_items)

[2.39, 3.39, 3.39, 2.39, 8.49, 10.98, 1.69, 11.75, 9.25, 9.25, 4.45, 8.75, 8.75, 11.25, 4.45, 2.39, 8.49, 8.49, 1.09, 8.75, 4.45, 8.99, 3.39, 10.98, 3.39, 2.39, 8.49, 8.99, 1.09, 8.49, 2.39, 8.99, 1.69, 8.99, 1.09, 8.75, 8.75, 4.45, 2.95, 11.75, 2.15, 4.45, 11.25, 11.75, 8.75, 10.98, 8.99, 3.39, 8.99, 3.99]
['Chips and Fresh Tomato Salsa' 'Izze' 'Nantucket Nectar'
 'Chips and Tomatillo-Green Chili Salsa' 'Chicken Bowl' 'Side of Chips'
 'Steak Burrito' 'Steak Soft Tacos' 'Chips and Guacamole'
 'Chicken Crispy Tacos' 'Chicken Soft Tacos' 'Chicken Burrito'
 'Canned Soda' 'Barbacoa Burrito' 'Carnitas Burrito' 'Carnitas Bowl'
 'Bottled Water' 'Chips and Tomatillo Green Chili Salsa' 'Barbacoa Bowl'
 'Chips' 'Chicken Salad Bowl' 'Steak Bowl' 'Barbacoa Soft Tacos'
 'Veggie Burrito' 'Veggie Bowl' 'Steak Crispy Tacos'
 'Chips and Tomatillo Red Chili Salsa' 'Barbacoa Crispy Tacos'
 'Veggie Salad Bowl' 'Chips and Roasted Chili-Corn Salsa'
 'Chips and Roasted Chili Corn Salsa' 'Carnitas Soft Tacos'

In [135]:
for i in range(len(nombre_items)):
    lista_nombres.append(df['item_name'][i])
    print(df['item_name'][i], df['valor_prod'][i])

Chips and Fresh Tomato Salsa 2.39
Izze 3.39
Nantucket Nectar 3.39
Chips and Tomatillo-Green Chili Salsa 2.39
Chicken Bowl 8.49
Chicken Bowl 10.98
Side of Chips 1.69
Steak Burrito 11.75
Steak Soft Tacos 9.25
Steak Burrito 9.25
Chips and Guacamole 4.45
Chicken Crispy Tacos 8.75
Chicken Soft Tacos 8.75
Chicken Bowl 11.25
Chips and Guacamole 4.45
Chips and Tomatillo-Green Chili Salsa 2.39
Chicken Burrito 8.49
Chicken Burrito 8.49
Canned Soda 1.09
Chicken Bowl 8.75
Chips and Guacamole 4.45
Barbacoa Burrito 8.99
Nantucket Nectar 3.39
Chicken Burrito 10.98
Izze 3.39
Chips and Fresh Tomato Salsa 2.39
Chicken Bowl 8.49
Carnitas Burrito 8.99
Canned Soda 1.09
Chicken Burrito 8.49
Chips and Tomatillo-Green Chili Salsa 2.39
Steak Burrito 8.99
Side of Chips 1.69
Carnitas Bowl 8.99
Bottled Water 1.09
Chicken Soft Tacos 8.75
Chicken Soft Tacos 8.75
Chips and Guacamole 4.45
Chips and Tomatillo Green Chili Salsa 2.95
Barbacoa Bowl 11.75
Chips 2.15
Chips and Guacamole 4.45
Chicken Bowl 11.25
Steak Burrit

In [136]:
valor_productos_nombre= pd.Series(lista_valores, index=nombre_items)

In [137]:
DFx= pd.DataFrame({'valor_productos_nombre': valor_productos_nombre})

In [138]:
DFx.sort_values(by='valor_productos_nombre', ascending= False)

valor_productos_nombre
Crispy Tacos                                            11.75
Veggie Soft Tacos                                       11.75
Steak Soft Tacos                                        11.75
Chips and Mild Fresh Tomato Salsa                       11.25
Barbacoa Burrito                                        11.25
Barbacoa Salad Bowl                                     10.98
Side of Chips                                           10.98
Veggie Burrito                                          10.98
Chicken Crispy Tacos                                     9.25
Chips and Guacamole                                      9.25
Canned Soft Drink                                        8.99
Barbacoa Crispy Tacos                                    8.99
Steak Bowl                                               8.99
Salad                                                    8.99
Veggie Salad                                             8.99
Carnitas Soft Tacos                                      8.99
6 Pack Soft Drink                                        8.75
Carnitas Salad Bowl                                      8.75
Chips                                                    8.75
Canned Soda                                              8.75
Chicken Burrito                                          8.75
Chips and Tomatillo-Red Chili Salsa                      8.75
Chips and Tomatillo Green Chili Salsa                    8.49
Chips and Tomatillo Red Chili Salsa                      8.49
Chicken Bowl                                             8.49
Chips and Roasted Chili-Corn Salsa                       8.49
Bottled Water                                            8.49
Chicken Salad Bowl                                       4.45
Carnitas Burrito                                         4.45
Steak Salad                                              4.45
Chicken Soft Tacos                                       4.45
Bowl                                                     4.45
Carnitas Salad                                           3.99
Izze                                                     3.39
Veggie Bowl                                              3.39
Barbacoa Soft Tacos                                      3.39
Veggie Crispy Tacos                                      3.39
Nantucket Nectar                                         3.39
Burrito                                                  2.95
Chips and Fresh Tomato Salsa                             2.39
Chips and Roasted Chili Corn Salsa                       2.39
Carnitas Bowl                                            2.39
Chips and Tomatillo-Green Chili Salsa                    2.39
Steak Crispy Tacos                                       2.39
Carnitas Crispy Tacos                                    2.15
Chicken Salad                                            1.69
Steak Burrito                                            1.69
Steak Salad Bowl                                         1.09
Veggie Salad Bowl                                        1.09
Barbacoa Bowl                                            1.09

CORRECCION: 

In [139]:
df.sort_values(by= "item_name", ascending= False)

order_id  quantity          item_name  \
3889      1559       2.0  Veggie Soft Tacos   
2851      1132       1.0  Veggie Soft Tacos   
738        304       1.0  Veggie Soft Tacos   
1395       567       1.0  Veggie Soft Tacos   
2384       948       1.0  Veggie Soft Tacos   
...        ...       ...                ...   
1916       774       1.0  6 Pack Soft Drink   
1124       465       1.0  6 Pack Soft Drink   
3789      1517       1.0  6 Pack Soft Drink   
1392       566       1.0  6 Pack Soft Drink   
3614      1448       1.0  6 Pack Soft Drink   

                                     choice_description  item_price  \
3889  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...       16.98   
2851  [Roasted Chili Corn Salsa (Medium), [Black Bea...        8.49   
738   [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25   
1395  [Fresh Tomato Salsa (Mild), [Pinto Beans, Rice...        8.49   
2384  [Roasted Chili Corn Salsa, [Fajita Vegetables,...        8.75   
...                                                 ...         ...   
1916                                        [Diet Coke]        6.49   
1124                                             [Coke]        6.49   
3789                                        [Diet Coke]        6.49   
1392                                        [Diet Coke]        6.49   
3614                                           [Nestea]        6.49   

      valor_prod  
3889        8.49  
2851        8.49  
738        11.25  
1395        8.49  
2384        8.75  
...          ...  
1916        6.49  
1124        6.49  
3789        6.49  
1392        6.49  
3614        6.49  

[4622 rows x 6 columns]

In [140]:
df.sort_values(by= ["item_name", "valor_prod"], ascending= [True, False])

order_id  quantity          item_name  \
298        129       1.0  6 Pack Soft Drink   
341        148       1.0  6 Pack Soft Drink   
357        154       1.0  6 Pack Soft Drink   
388        168       1.0  6 Pack Soft Drink   
417        182       1.0  6 Pack Soft Drink   
...        ...       ...                ...   
781        322       1.0  Veggie Soft Tacos   
2384       948       1.0  Veggie Soft Tacos   
1395       567       1.0  Veggie Soft Tacos   
2851      1132       1.0  Veggie Soft Tacos   
3889      1559       2.0  Veggie Soft Tacos   

                                     choice_description  item_price  \
298                                            [Sprite]        6.49   
341                                         [Diet Coke]        6.49   
357                                              [Coke]        6.49   
388                                         [Diet Coke]        6.49   
417                                         [Diet Coke]        6.49   
...                                                 ...         ...   
781   [Fresh Tomato Salsa, [Black Beans, Cheese, Sou...        8.75   
2384  [Roasted Chili Corn Salsa, [Fajita Vegetables,...        8.75   
1395  [Fresh Tomato Salsa (Mild), [Pinto Beans, Rice...        8.49   
2851  [Roasted Chili Corn Salsa (Medium), [Black Bea...        8.49   
3889  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...       16.98   

      valor_prod  
298         6.49  
341         6.49  
357         6.49  
388         6.49  
417         6.49  
...          ...  
781         8.75  
2384        8.75  
1395        8.49  
2851        8.49  
3889        8.49  

[4622 rows x 6 columns]

In [141]:
df.sort_values(by= ["item_name", "valor_prod"], ascending= [True, False]).tail(20)

order_id  quantity          item_name  \
2156       869       1.0  Veggie Salad Bowl   
3223      1289       1.0  Veggie Salad Bowl   
4109      1646       1.0  Veggie Salad Bowl   
4201      1677       1.0  Veggie Salad Bowl   
4261      1700       1.0  Veggie Salad Bowl   
960        394       1.0  Veggie Salad Bowl   
1316       536       1.0  Veggie Salad Bowl   
2223       896       1.0  Veggie Salad Bowl   
2269       913       1.0  Veggie Salad Bowl   
2683      1066       1.0  Veggie Salad Bowl   
3293      1321       1.0  Veggie Salad Bowl   
4541      1805       1.0  Veggie Salad Bowl   
4573      1818       1.0  Veggie Salad Bowl   
738        304       1.0  Veggie Soft Tacos   
1699       688       1.0  Veggie Soft Tacos   
781        322       1.0  Veggie Soft Tacos   
2384       948       1.0  Veggie Soft Tacos   
1395       567       1.0  Veggie Soft Tacos   
2851      1132       1.0  Veggie Soft Tacos   
3889      1559       2.0  Veggie Soft Tacos   

                                     choice_description  item_price  \
2156  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25   
3223  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25   
4109  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25   
4201  [Fresh Tomato Salsa, [Fajita Vegetables, Black...       11.25   
4261  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25   
960   [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75   
1316  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...        8.75   
2223      [Roasted Chili Corn Salsa, Fajita Vegetables]        8.75   
2269  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...        8.75   
2683  [Roasted Chili Corn Salsa, [Fajita Vegetables,...        8.75   
3293  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...        8.75   
4541  [Tomatillo Green Chili Salsa, [Fajita Vegetabl...        8.75   
4573  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75   
738   [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25   
1699  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25   
781   [Fresh Tomato Salsa, [Black Beans, Cheese, Sou...        8.75   
2384  [Roasted Chili Corn Salsa, [Fajita Vegetables,...        8.75   
1395  [Fresh Tomato Salsa (Mild), [Pinto Beans, Rice...        8.49   
2851  [Roasted Chili Corn Salsa (Medium), [Black Bea...        8.49   
3889  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...       16.98   

      valor_prod  
2156       11.25  
3223       11.25  
4109       11.25  
4201       11.25  
4261       11.25  
960         8.75  
1316        8.75  
2223        8.75  
2269        8.75  
2683        8.75  
3293        8.75  
4541        8.75  
4573        8.75  
738        11.25  
1699       11.25  
781         8.75  
2384        8.75  
1395        8.49  
2851        8.49  
3889        8.49

### 6. Cuál es la cantitad del producto más caro?

In [142]:
df['valor_prod'].argmax()

281

In [143]:
df.iloc[281]

order_id                                                            123
quantity                                                            2.0
item_name                                              Steak Salad Bowl
choice_description    [Tomatillo Red Chili Salsa, [Black Beans, Chee...
item_price                                                        23.78
valor_prod                                                        11.89
Name: 281, dtype: object

In [144]:
# el producto mas caro es steak salad bowl

In [145]:
len(df['item_name'][df['item_name'] == "Steak Salad Bowl"])

29

In [146]:
SSB= df['quantity'][df['item_name'] == "Steak Salad Bowl"]

In [147]:
SSB.sum()

31.0

In [148]:
# se han comprado un total de de 31 steak salad bowls [(27*1) + (2*2)]

CORRECCION:

In [149]:
max_value_productos= df[df['valor_prod'] == df['valor_prod'].max()]['item_name'].unique()

In [150]:
df[df['valor_prod'] == df['valor_prod'].max()]['quantity'].sum()

30.0

se han pedido 30 ensaladas en las que se ha pagado el precio máximo

vale, el producto más caro es la ensalada, pero es por el precio de la ensalada en sí o por el precio de la ensalada más los toppings? es decir, cual es el precio basico del producto?

In [151]:
df[df['item_name'].isin(max_value_productos)]['quantity'].sum()

47.0

Ensaladas en total se han pedido 47, pero no todas ellas tienen ese precio más caro. El precio básico del producto no es igual al más caro

### 7. Cuáles han sido las cantidades vendidas de Chicken Bowl? Y cuántos pedidos?

In [152]:
CB= df['quantity'][df['item_name'] == "Chicken Bowl"]

In [153]:
CB.sum()

761.0

In [154]:
# se han vendido 761 unidades de Chicken Bowl

In [155]:
CB_pedidos= pd.Series(df['order_id'][df['item_name'] == "Chicken Bowl"])

In [156]:
len(CB_pedidos)

726

In [157]:
CB_pedidos.nunique()

615

In [158]:
#ha habido 615 pedidos de chicken bowl

ESTE ESTA BIEN

### 8. Cuántas veces alguien ha pedido más de una Canned Soda? Piénsalo bien.

In [159]:
CS_pedidos= pd.Series(df['order_id'][df['item_name'] == "Canned Soda"][df['quantity'] > 1])

In [160]:
CS_pedidos.nunique()

18

In [161]:
# 18 veces se han pedido mas de una Canned Soda

ESTA BIEN
Otra forma de hacerlo:

In [162]:
len(df[(df['item_name'] == "Canned Soda") & (df['quantity']>1)])

20

In [163]:
df[(df['item_name'] == "Canned Soda") & (df['quantity']>1)]['order_id'].nunique()

18

CORRECCION: Rizar el rizo.

En un mismo pedido, puede haber varias lineas de pedidos con un Canned Soda. En este caso, la cantidad de Canned Soda en esa linea seria =1, pero se habrían pedido >1 en un mismo pedido. Habria que agrupar primero por pedido y luego filtrar el pedido.

In [164]:
df[(df['item_name'] == "Canned Soda")].groupby('order_id')['quantity'].sum() 

order_id
9       2.0
14      1.0
23      2.0
24      1.0
47      1.0
       ... 
1562    1.0
1575    1.0
1578    1.0
1604    1.0
1621    1.0
Name: quantity, Length: 94, dtype: float64

In [165]:
pedidos_refrescos= df[(df['item_name'] == "Canned Soda")].groupby('order_id')['quantity'].sum() 

In [166]:
len(pedidos_refrescos[pedidos_refrescos>1])

24